flow/flow.go

```diff
@@ -363,13 +367,68 @@ func (f *Flow) Init(key string, now int64, packet *gopacket.Packet, length int64
 	f.UpdateUUID(key, L2ID, L3ID)
 }
 
-// Update a flow metrics
+// Update a flow metrics and latency
 func (f *Flow) Update(now int64, packet *gopacket.Packet, length int64) {
 	f.Last = now
 
 	if updated := f.updateMetricsWithLinkLayer(packet, length); !updated {
 		f.updateMetricsWithNetworkLayer(packet)
 	}
+
+	f.updateLatency(packet)
+}
+
+func (f *Flow) updateLatency(packet *gopacket.Packet) {
+	if f.Link != nil && f.Link.latencySet == false {
+		ethernetLayer := (*packet).Layer(layers.LayerTypeEthernet)
+		ethernetPacket, ok := ethernetLayer.(*layers.Ethernet)
+		if ok && (f.Link.A == ethernetPacket.DstMAC.String()) {
+			f.Link.Latency = (*packet).Metadata().Timestamp.UnixNano() - f.Link.Latency
+			f.Link.latencySet = true
+		}
+	}
```

Comment:
Am I right that in the patch we assume that since the two sequential packets from two different directions are first in the flow - they must be request/reply? I am not sure if such assumption is valid... The "reply" might be something belonging to an old session within the same flow. Or for example A sends to B via two different TCP sockets SYN requests, then B replies to one of them and looses one of them. On the layer of Ethernet - what will we keep as a Latency? 
I do understand capturing first packets in TCP layer, but still the possible scenario is that A sends SYN and SYN and B replies. Still, we cannot be sure we receive SYN/ACK from B to the first or to the second packet from A. 
I would say that it is more accurate and informative to keep time-stemp for first A->B packet and first B->A packet. Still, I am not sure if for all layers it is valid to calculate based on it the latency.
General comment, I would call to it RTT rather than Latency, but maybe I miss something...
I think we should keep the whole first packet in both directions because it might give us a lot of info for diagnostics (at least we can do it as an option).
Please, tell me what you think about it because my patch in progress that captures SYN, SYN/ACK packets looks like related to your patch, so I will be glad to discuss it and come to some consensus :).
Thank you,
Lera